src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/AuditService.kt

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.service
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest
+import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.config.AuthenticationFacade
+import uk.gov.justice.hmpps.sqs.HmppsQueue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import java.time.Instant
+
+@Service
+class AuditService(
+  @Value("\${spring.application.name}")
+  private val serviceName: String,
+  private val hmppsQueueService: HmppsQueueService,
+  private val telemetryClient: TelemetryClient,
+  private val objectMapper: ObjectMapper,
+  private val authenticationFacade: AuthenticationFacade,
+) {
+  private val auditQueue by lazy { hmppsQueueService.findByQueueId("audit") as HmppsQueue }
+  private val auditSqsClient by lazy { auditQueue.sqsClient }
+  private val auditQueueUrl by lazy { auditQueue.queueUrl }
+
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun sendMessage(auditType: AuditType, id: String, details: Any, username: String? = null) {
+    val auditEvent = AuditEvent(
+      what = auditType.name,
+      who = username ?: authenticationFacade.currentUsername ?: "non-associations-api",
+      service = serviceName,
+      details = objectMapper.writeValueAsString(details),
```

Comment:
I see there is a `toJson()` method added to `Any`, does it mean you could use it on `details` too here?

e.g. `details = details.toJson()`. It would probably read more nicely ☺️

Reply:
Done